configs/react/index.js

```diff
@@ -11,55 +11,64 @@ module.exports = {
-    "react/button-has-type": "error",
-    "react/display-name": "off",
-    "react/jsx-boolean-value": "error",
-    "react/jsx-curly-brace-presence": ["error", { props: "never", children: "never" }],
-    "react/jsx-filename-extension": ["error", { extensions: [".jsx", ".tsx"] }],
-    "react/jsx-wrap-multilines": "off",
-    "react/no-array-index-key": "error",
-    "react/no-this-in-sfc": "error",
-    "react/no-unused-prop-types": "error",
-    "react/no-unused-state": "error",
-    "react/void-dom-elements-no-children": "error",
+    'react/jsx-props-no-spreading': 'off',
+    'react/react-in-jsx-scope': 'off', // no longer needed with the new JSX transform
+    'react-hooks/exhaustive-deps': 'error',
+    'react/button-has-type': 'error',
+    'react/display-name': 'off',
+    'react/jsx-boolean-value': 'error',
+    'react/jsx-curly-brace-presence': ['error', { props: 'never', children: 'never' }],
+    'react/jsx-filename-extension': ['error', { extensions: ['.jsx', '.tsx'] }],
+    'react/jsx-wrap-multilines': 'off',
+    'react/no-array-index-key': 'error',
+    'react/no-this-in-sfc': 'error',
+    'react/no-unused-prop-types': 'error',
+    'react/no-unused-state': 'error',
+    'react/void-dom-elements-no-children': 'error',
   },
   overrides: [
     {
-      files: ["*.tsx"],
+      files: ['*.tsx'],
       rules: {
-        "react/prop-types": "off",
-        "@typescript-eslint/naming-convention": [
-          "error",
+        'react/prop-types': 'off',
+        '@typescript-eslint/naming-convention': [
+          'error',
           {
-            selector: "default",
-            format: ["strictCamelCase"],
-            leadingUnderscore: "forbid",
-            trailingUnderscore: "forbid",
+            selector: 'default',
+            format: ['strictCamelCase'],
+            leadingUnderscore: 'forbid',
+            trailingUnderscore: 'forbid',
           },
           {
-            selector: "typeLike",
-            format: ["StrictPascalCase"],
+            selector: 'objectLiteralProperty',
+            modifiers: ['requiresQuotes'],
+            format: null,
           },
           {
-            selector: "variable",
-            // Exception for FunctionComponents
-            format: ["strictCamelCase", "StrictPascalCase", "UPPER_CASE"],
+            selector: 'typeLike',
+            format: ['StrictPascalCase'],
           },
           {
-            selector: "function",
+            selector: 'function',
             // Exception for FunctionComponents
-            format: ["strictCamelCase", "StrictPascalCase"],
+            format: ['strictCamelCase', 'StrictPascalCase'],
+          },
+          {
+            selector: 'variable',
+            format: ['strictCamelCase', 'UPPER_CASE'],
+          },
+          {
+            selector: 'variable',
+            modifiers: ['exported'],
+            format: ['strictCamelCase', 'StrictPascalCase', 'UPPER_CASE'],
```

Comment:
In our React standard, aren't all JSX components expected to be created as normal functions, and not arrows > vars?

Or do we need this for when `memo` is used?

Should we add a comment for the exact reasons in the code, so we know why we made this config later? :)

Reply:
> Or do we need this for when `memo` is used?

yes for HOC's and for contexts, e.g. https://github.com/mediamonks/pota/blob/main/templates/react/src/stores/global.tsx#L4

_Added an explainer comment_